internal/stolonctl/pkg/cluster/upgrade.go

```diff
@@ -84,59 +98,297 @@ func Upgrade(ctx context.Context, config Config) error {
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	return res, nil
+}
+
+func (u *upgradeControl) backupPostgres() error {
+	if err := u.createPgPassFile(); err != nil {
 		return trace.Wrap(err)
 	}
-	options := []string{fmt.Sprintf("-h%s", config.Postgres.Host),
-		fmt.Sprintf("-p%s", config.Postgres.Port),
-		fmt.Sprintf("-U%s", config.Postgres.User), "-c",
-		fmt.Sprintf("-f%s", config.Postgres.BackupPath)}
+	options := []string{fmt.Sprintf("-h%s", u.config.Postgres.Host),
+		fmt.Sprintf("-p%s", u.config.Postgres.Port),
+		fmt.Sprintf("-U%s", u.config.Postgres.User), "-c",
+		fmt.Sprintf("-f%s", u.config.Postgres.BackupPath)}
 
 	cmd := exec.Command(pgDumpCommand, options...)
 	env := os.Environ()
 	// Add PGPASSFILE env variable to set path to the password file
-	env = append(env, fmt.Sprintf("PGPASSFILE=%s", config.Postgres.PgPassPath))
+	env = append(env, fmt.Sprintf("PGPASSFILE=%s", u.config.Postgres.PgPassPath))
 	cmd.Env = env
 
 	output, err := utils.Run(cmd)
 	if err != nil {
 		return trace.Wrap(err, "Command output: %v", string(output))
 	}
-	log.Infof("Backed up Postgres to %s", config.Postgres.BackupPath)
+	log.Infof("Backed up Postgres to %s", u.config.Postgres.BackupPath)
 
 	return nil
 }
 
-func createPgPassFile(config Config) error {
-	content := fmt.Sprintf("%s:%s:%s:%s:%s", config.Postgres.Host, config.Postgres.Port,
-		"*", config.Postgres.User, config.Postgres.Password)
+func (u *upgradeControl) createPgPassFile() error {
+	content := fmt.Sprintf("%s:%s:%s:%s:%s", u.config.Postgres.Host, u.config.Postgres.Port,
+		"*", u.config.Postgres.User, u.config.Postgres.Password)
 
-	return common.WriteFileAtomic(config.Postgres.PgPassPath, []byte(content), 0600)
+	return common.WriteFileAtomic(u.config.Postgres.PgPassPath, []byte(content), 0600)
 }
+
+// deleteDeployment deletes sentinels deployment
+func (u *upgradeControl) deleteDeployment(name string) error {
+	dsClient := u.kubeClient.Client.AppsV1().Deployments(u.config.Namespace)
+	deletePolicy := metav1.DeletePropagationForeground
+	err := dsClient.Delete(name, &metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	})
+	if err != nil {
+		return trace.Wrap(err, "cannot delete stolon-sentinel deployment")
+	}
+	return nil
+}
+
+// deleteDaemonset deletes keepers daemonset
+func (u *upgradeControl) deleteDaemonset(name string) error {
+	dsClient := u.kubeClient.Client.ExtensionsV1beta1().DaemonSets(u.config.Namespace)
+	deletePolicy := metav1.DeletePropagationForeground
+	err := dsClient.Delete(name, &metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
+	})
+	if err != nil {
+		return trace.Wrap(err, "cannot delete stolon-keeper daemonset")
+	}
+	return nil
+}
+
+func (u *upgradeControl) upgradePostgresSchema(ctx context.Context, pgRole, nodeName, id string) error {
+	jobControl, err := rigging.NewJobControl(rigging.JobConfig{u.generateJob(pgRole, nodeName, id), u.kubeClient.Client})
```

Comment:
nit: this could use a bit of formatting for readability:
```go
jobControl, err := rigging.NewJobControl(
	rigging.JobConfig{
		Job:       u.generateJob(pgRole, nodeName, id),
		Clientset: u.kubeClient.Client,
	},
)
```